app/matlab_reader.cpp

```diff
@@ -0,0 +1,433 @@
+        } else if (cm[1].matched) {
+            BOOST_LOG_TRIVIAL(trace) << "Matched stream and frame: " << p.path().string() << " at " << path.string() << ": " << cm[1] << ", " << cm[2] << ", " << cm[3];
+            channel = cm[1];
+            stream = std::stoi(cm[2]);
+            frame = std::stoi(cm[3]);
+
+            auto& a = aggSF[channel];
+            a.addSuffix(suffix);
+            a.addStream(stream);
+            a.addFrame(frame);
+        } else {
+            BOOST_LOG_TRIVIAL(info) << "Ignoring path " << p.path().string() << ", does not match any expected file name.";
+            _ignoredPaths.push_back(p.path());
+            continue;
+        }
+        BOOST_LOG_TRIVIAL(trace) << "Adding frame file " << path.string() << " from " << p.path().string();
+        _files.frames[ElementKey(stream, frame, channel)].insert(path);
+    }
+    StreamNumber firstStream = std::numeric_limits<StreamNumber>::max();
+    StreamNumber lastStream = -1;
+    FrameNumber firstFrame = std::numeric_limits<StreamNumber>::max();
+    FrameNumber lastFrame = -1;
+    for(const auto& kv : aggSF){
+        if(EXPECTED_CHANNELS.find(kv.first) == EXPECTED_CHANNELS.end()){
+            _ignoredChannels.push_back(kv.first);
+            continue;
+        }
+        firstStream = std::min(firstStream, kv.second.firstStream);
+        lastStream = std::max(lastStream, kv.second.lastStream);
+        firstFrame = std::min(firstFrame, kv.second.firstFrame);
+        lastFrame = std::max(lastFrame, kv.second.lastFrame);
+    }
+
+    for(const auto& kv : aggF){
+        if(EXPECTED_CHANNELS.find(kv.first) == EXPECTED_CHANNELS.end()){
+            _ignoredChannels.push_back(kv.first);
+        }
+        firstFrame = std::min(firstFrame, kv.second.firstFrame);
+        lastFrame = std::max(lastFrame, kv.second.lastFrame);
+    }
+
+    _valid = firstStream <= lastStream && firstFrame <= lastFrame;
+
+    if(!_valid){
+        _beginStream = 0;
+        _endStream = 0;
+        _beginFrame = 0;
+        _endFrame = 0;
+        return;
+    } else {
+        _beginStream = firstStream;
+        _endStream = lastStream + 1;
+        _beginFrame = firstFrame;
+        _endFrame = lastFrame + 1;
+    }
+
+    _cache.rotationCorrections = readRotationCorrections();
+    _cache.camchain = readCamchain();
+}
+
+
+bool MatlabReader::valid() const {
+    return _valid;
+}
+
+StreamNumber MatlabReader::beginStream() const {
+    return _beginStream;
+}
+
+StreamNumber MatlabReader::endStream() const {
+    return _endStream;
+}
+
+FrameNumber MatlabReader::beginFrame() const {
+    return _beginFrame;
+}
+
+FrameNumber MatlabReader::endFrame() const {
+    return _endFrame;
+}
+
+const std::vector<fs::path>& MatlabReader::ignoredPaths() const {
+    return _ignoredPaths;
+}
+
+const std::vector<std::string>& MatlabReader::ignoredChannels() const {
+    return _ignoredChannels;
+}
+
+std::vector<Frame> MatlabReader::frames(FrameNumber f) const {
```

Comment:
can this method throw? should probably have a `nothrow`, maybe a valid flag on Frame?